packages/rpc-methods/src/restricted/manageState.ts

```diff
@@ -104,22 +109,20 @@ export const STORAGE_SIZE_LIMIT = 104857600; // In bytes (100MB)
  * @returns The method implementation which either returns `null` for a successful state update/deletion or returns the decrypted state.
  * @throws If the params are invalid.
  */
-function getManageStateImplementation({
+export function getManageStateImplementation({
   clearSnapState,
   getSnapState,
   updateSnapState,
 }: ManageStateMethodHooks) {
   return async function manageState(
-    options: RestrictedMethodOptions<
-      [ManageStateOperation, Record<string, Json>]
-    >,
+    options: RestrictedMethodOptions<ManageStateArgs>,
   ): Promise<null | Record<string, Json>> {
     const {
-      params = [],
+      params = {},
       method,
       context: { origin },
     } = options;
-    const [operation, newState] = params;
+    const { operation, newState } = getValidatedParams(params);
 
     switch (operation) {
       case ManageStateOperation.clearState:
```

Comment:
Should we move the validation specific to `updateState` to `getValidatedParams`?